hopp/simulation/technologies/wind/pysam_wind.py

```diff
@@ -0,0 +1,65 @@
+from attrs import define, field
+
+import PySAM.Windpower as Windpower
+
+from hopp.simulation.base import BaseClass
+from hopp.simulation.technologies.sites import SiteInfo
+from hopp.type_dec import resource_file_converter
+from hopp.utilities.utilities import load_yaml
+
+
+@define
+class PySAMWind(BaseClass):
+    config_dict: dict = field(converter=dict)
+    site: SiteInfo = field()
+    timestep: tuple = field(default=(), converter=tuple)
+
+    system_model: Windpower = field(init=False)
+
+    def __attrs_post_init__(self):
+        input_file_path = resource_file_converter(self.config_dict["simulation_input_file"])
+        filetype = input_file_path.suffix.strip(".")
+        if filetype.lower() in ("yml", "yaml"):
+            input_dict = load_yaml(input_file_path)
+        else:
+            raise ValueError("Supported import filetype is YAML")
+
+        self.system_model = Windpower.new()
+        self.system_model.assign(input_dict)
+
+        self.wind_turbine_rotor_diameter = input_dict['Turbine']['wind_turbine_rotor_diameter']
+        self.wind_farm_xCoordinates = input_dict['Farm']['wind_farm_xCoordinates']
+        self.wind_farm_yCoordinates = input_dict['Farm']['wind_farm_yCoordinates']
+        self.nTurbs = len(self.wind_farm_xCoordinates)
+        self.system_model.value("wind_resource_data", self.site.wind_resource.data)
+
+        # turbine power curve (array of kW power outputs)
+        self.wind_turbine_powercurve_powerout = [1] * self.nTurbs
+
+    def value(self, name: str, set_value=None):
+        """
+        if set_value = None, then retrieve value; otherwise overwrite variable's value
+        """
+        if set_value:
+            self.__setattr__(name, set_value)
+        else:
+            return self.__getattribute__(name)
+
+    def execute(self, project_life):
+        self.system_model.execute(project_life)
+
+    @property
```

Comment:
Instead of explicitly making an interface for these properties (which are only a subset of all the available ones that a user may be interested in), is it possible to automatically create them, or just pass the accessor directly to the system model. Not sure if it's possible but if `pysam_wind.annual_energy` could just return `wind_source.annual_energy`?

On a high level, I'm not sure what this new layer pysam_wind is supposed to achieve besides having the attr_post_init function that allows recursive class instantiation. If that's the point, then I think trying to make it as minimal as possible would be good, so passing all `set` and `get` to the underlying model without intermediate functions and intermediate data. But again, I'm not sure what the intention is here

Reply:
I look at having power source inherit from base class and see if that allows these extra properties to be removed.